sdk/vppagent/dataplane/chained_dataplane_server.go

```diff
@@ -0,0 +1,45 @@
+package dataplane
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/crossconnect"
+	"github.com/networkservicemesh/networkservicemesh/dataplane/pkg/apis/dataplane"
+	"github.com/networkservicemesh/networkservicemesh/sdk/vppagent/dataplane/state"
+)
+
+type ChainedDataplaneServer interface {
+	dataplane.DataplaneServer
+}
+
+type chainedDataplaneServer struct {
+	handlers []dataplane.DataplaneServer
+}
+
+func (c *chainedDataplaneServer) Request(ctx context.Context, crossConnect *crossconnect.CrossConnect) (*crossconnect.CrossConnect, error) {
+	nextCtx := state.WithChain(ctx, c, c.handlers)
+	return state.NextDataplaneRequest(nextCtx, crossConnect)
```

Comment:
Rather than NextDataplaneRequest(...) and NextDataplaneClose(...) ... lets just do Next().Request(...) or Next(...).Close().

Reply:
thanks, fixed